src/games.c

```diff
@@ -1691,7 +1691,47 @@ int igraph_callaway_traits_game (igraph_t *graph, igraph_integer_t nodes,
     igraph_real_t maxcum;
     igraph_vector_t nodetypes;
 
-    /* TODO: parameter checks */
+    /* Argument contracts */
+    if(nodes < 0){
+        IGRAPH_ERROR("The number of vertices must be non-negative", IGRAPH_EINVAL);
+    }
+    
+    if (types < 1) {
+        IGRAPH_ERROR("The number of vertex types must be at least 1", IGRAPH_EINVAL);
+    }
+
+    if (igraph_vector_size(type_dist) != types) {
+        IGRAPH_ERROR("The vertex type distribution vector must agree in length with the number of types", IGRAPH_EINVAL);
+    }
+    
+    if (igraph_matrix_size(pref_matrix) != types * types) {
+        IGRAPH_ERROR("Size of matrix is not consistent to the vertex type distribution vector", IGRAPH_EINVAL);
+    }
+    if(igraph_matrix_nrow(pref_matrix) != igraph_matrix_ncol(pref_matrix)){
+                IGRAPH_ERROR("The matrix must be square", IGRAPH_EINVAL);
+    }
+    
+    for (i = 0; i < types; i++) {
+        if(VECTOR(*type_dist)[i] < 0){
+            IGRAPH_ERROR("Negative component in the vertex type distribution vector", IGRAPH_EINVAL);
+        }
+        igraph_vector_t row_i;
+        igraph_matrix_get_row(pref_matrix, &row_i, i); 
+        for(j = 0; j < types; j++){
+            if(VECTOR(row_i)[j] < 0 || VECTOR(row_i)[j] > 1){
```

Comment:
We have `igraph_matrix_minmax()` to retrieve the smallest and largest elements of the matrix; please use that to test whether all elements of the matrix fall between 0 and 1.